src/wp-includes/class-wp-theme-json.php

```diff
@@ -0,0 +1,375 @@
+<?php
+/**
+ * Process of structures that adhere to the theme.json schema.
+ *
+ * @package WordPress
+ */
+
+/**
+ * Class that encapsulates the processing of
+ * structures that adhere to the theme.json spec.
+ */
+class WP_Theme_JSON {
+
+	/**
+	 * Container of data in theme.json format.
+	 *
+	 * @var array
+	 */
+	private $theme_json = null;
+
+	const VALID_TOP_LEVEL_KEYS = array(
+		'version',
+		'settings',
+	);
+
+	const VALID_SETTINGS = array(
+		'border'     => array(
```

Comment:
There're a few things here that are not yet stable (same in styles). I think this is a good opportunity to add a specific hook: from the plugin, we may want to add new allowed settings & styles for WordPress core to process normally. While an advanced change such as theme.json format change, merge algorithm, etc. is a bit more complex and can be done by hooking into the existing block editor filter ([see](https://github.com/WordPress/wordpress-develop/pull/1262/files#r634596983)), a simple thing like this should also be possible.

Reply:
1a84edbf55ee1cafe31848f3bf298d7da60caecb